### PR TITLE
Improve CLI ergonomy

### PR DESCRIPTION
**Pull request by caglaryucekaya**
Solves #5 

- [ ]  Compress file x.y to x.y.bz2 and decompress file x.y.bz2 to x.y
- [ ]  Check if the output files already exist and fail with an error message 
- [ ]  Handle I/O failures with error messages

### CODE REVIEW

**Comment by torfmaster** [comment]
Hi, thank you for your PR! I have some additional ideas
* you could create an error enum that on one hand side represent the domain errors (output file already exists) and the io errors
* you could implement display for the error variants instead of panicking immediately
* you could refactor the generation of the input file and output file in functions that return the specific error type and do the error handling on "top level" (and use the `?` operator to exclude error handling from the normal control flow).

Moreover, the integration tests fail. You can fix the tests by deleting output/input files in `tests.sh`.

If you have questions, feel free to ask!

**Comment by caglaryucekaya** [comment]
Thanks for the suggestions! I made some changes accordingly and need further comments to see if there are any problems, and what I can do more.

File: cli/src/main.rs [hunk 1]
```diff
@@ -33,15 +34,57 @@ pub(crate) enum EncodingOptions {
...
+    let file = File::create(&file_path)?;
+    Ok(file)
+}
+
 fn main() {
     let opt = Opt::from_args();
     match opt {
         Opt::Decompress { input } => {
             for file_name in input {
+                let mut in_file = File::open(&file_name).unwrap_or_else(|err| {
```
Comment (torfmaster) [inline cli/src/main.rs]: In order to remove the duplication which happens due to error handling you could refactor the whole block into a function which returns an appropriate `Result`. Then you can do a single `.expect` or `.unwrap_or_else` on the result (and you could use the `?` operator inside the function.

Comment (torfmaster) [inline cli/src/main.rs]: In order to remove the duplication which happens due to error handling you could refactor the whole block into a function which returns an appropriate `Result`. Then you can do a single `.expect` or `.unwrap_or_else` on the result (and you could use the `?` operator inside the function.

**Comment by caglaryucekaya** [comment]
Is it alright now?

**Comment by torfmaster** [comment]
Hi, it looks fine. My ultimate design goal is that all error handling (i.e. printing error messages and exiting) is happening at one central place in main and all the other places are free of error handling (except for using the `?` operator of course). I'm planning to do this eventually when I am doing error handling for decompression but if you want to take further steps in that direction, feel free. As you wish, I can either merge the PR or you take some steps in this direction.

**Comment by caglaryucekaya** [comment]
How about something like this? I propagated all errors from a try_main function to the higher-level main function, which is only for error handling. In this way all error handling is in one place. I can change it back in case this is not what you wanted.

**Comment by torfmaster** [comment]
I'm impressed this looks pretty good and I'm going to merge this! Thank you for your contribution!

**Comment by caglaryucekaya** [comment]
Thanks :)

**Pull request merged**